src/cpp/session/modules/SessionGit.cpp

```diff
@@ -204,10 +204,52 @@ std::string gitBin()
 }
 #endif
 
+bool waitForIndexLock(const FilePath& workingDir)
+{
+   using namespace boost::posix_time;
+   
+   // count the number of retries (avoid getting stuck in wait loops when
+   // an index.lock file exists and is never cleaned up)
+   static int retryCount = 0;
```

Comment:
Why is this static? In the case where we use up all the retries, it looks like the next attempt to wait for the lockfile to clear will start out over the limit and consequently not retry at all (let me know if I'm misreading something)